src/config.rs

```diff
@@ -173,6 +180,68 @@ impl StarshipConfig {
         }
     }
 
+    /// Load additional config values from the environment
+    /// Variables are prefixed with `STARSHIP_` and delimited with `__`
+    /// e.g. `STARSHIP_CONFIG__JAVA__DISABLED`
+    pub fn load_from_env(&mut self) {
+        for (key, value) in std::env::vars() {
+            if let Some(name) = key.strip_prefix("STARSHIP_CONFIG__") {
+                log::debug!("Loading config from environment: {name}={value}");
```

Comment:
I would suggest logging with the full environment variable name, for clarity:

```suggestion
                log::debug!("Loading config from environment: {key}={value}");
```